types/messages.go

```diff
@@ -4,28 +4,24 @@ import (
 	"bytes"
 	"encoding/binary"
 	"encoding/hex"
-	"fmt"
 
+	"github.com/attestantio/go-eth2-client/spec/phase0"
 	"github.com/pkg/errors"
 )
 
-// ValidatorPK is an eth2 validator public key
-type ValidatorPK []byte
+// ValidatorPK is an eth2 validator public key 48 bytes long
+type ValidatorPK phase0.BLSPubKey
+
+// ShareValidatorPK is a partial eth2 validator public key 48 bytes long
+type ShareValidatorPK []byte
```

Comment:
This is just for readability.

Before we had this field:
`SharePubKey     []byte`

So now instead of creating byte arrays you create `SharePubKey` arrays

Reply:
We are defining:
```go
type ValidatorPK phase0.BLSPubKey
type ShareValidatorPK []byte
```
However, I believe that `ShareValidatorPK` could also be of type `phase0.BLSPubKey`, no?

Reply:
The idea is to distinguish between a full validator key and a partial validator key

Reply:
Yup. What I mean is that both could be of type `phase0.BLSPubKey`, no?

Reply:
yeah...
but then you can't distinguish between them